File: LeetCode/Medium/0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        if (nums.size() < 3) return min(nums[0], nums[nums.size()-1]);
        
        int l = 0, r = nums.size() - 1, mid = 0, ret = nums[0];
        
        while (l <= r){
            mid = (l + r) / 2;
            
            if (nums[mid] > ret) l = mid + 1;
            else {
                r = mid - 1;
                ret = nums[mid];
            }
        }
        
        return ret;
    }
};

File: LeetCode/Medium/0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.go
func findMin(nums []int) int {
    if len(nums) < 3 { 
        if nums[0] <= nums[len(nums) - 1] { return nums[0] }
        return nums[1]
    }
    
    l := 0
    r := len(nums) - 1
    mid := 0
    ret := nums[0]
    
    for l <= r {
        mid = (l + r) / 2
        
        if nums[mid] > ret { 
            l = mid + 1
        } else {
            r = mid - 1
            ret = nums[mid]
        }
    }
    
    return ret
}